### DISCUSSION TITLE
"#define USE_TASMOTA_DISCOVERY"

### DISCUSSION THREAD

**Discussion by NsinghP** [arendst/Tasmota#12348] [Q&A] [answered]
Hello, 

I am compiling my .bin file with "#define USE_TASMOTA_DISCOVERY" option. I had originally used this line to test "Tasmomanager" https://github.com/arendst/Tasmota/discussions/11509 . The requirement was to compile without home assistant and use "#define USE_TASMOTA_DISCOVERY".  Since I am not using Tasmomamanger should I remove this line? or Is it needed for other features to work properly?

Regards,

Navi

**Comment by Jason2866** [comment, marked as answer]
No, it is not needed when not using a mqtt driven software which should recognize devices automatically.

**Reply by NsinghP** [reply]
Thanks

**Reply by sfromis** [reply]
The feature provides metadata published retained which can be very useful to backend tools wanting to know what the device is and what features it has. Just by subscribing to MQTT, and getting the initial messages, a tool can know a lot about all of your Tasmota devices, if they have that feature active.

**Reply by NsinghP** [reply]
>  a tool can know a lot about all of your Tasmota devices, if they have that feature active.

Ok Thanks, Then I will keep it. Just in case it comes in handy for some app.
